src/main/java/com/google/starfish/services/FavoriteNoteService.java

```diff
@@ -115,6 +138,77 @@ public long getNumFavoritesByNoteId(DataSource pool, long noteId) throws SQLExce
     }
   }
 
+  /** Gets trending notes based on number of favorites in a given timespan, filtered by school and/or course */
+  public Object[][] getTrendingNotesBySchoolOrCourse(DataSource pool, Recency recency, String school, String course) throws SQLException {
+    String filterStmt = getStatementToFilterBySchoolOrCourse(school, course);
+    Date date = getDateBasedOnRecency(recency);
+    List<Object> notes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
+      String stmt = getTrendingNotesStatement(filterStmt);
+      try (PreparedStatement selectStmt = conn.prepareStatement(stmt)) {
+        selectStmt.setDate(1, date);
+        ResultSet rs = selectStmt.executeQuery();
+        while (rs.next()) {
+          Note thisNote = constructNoteFromSqlResult(pool, rs);
+          Long numFavoritesInTimespan = rs.getLong(NUM_FAVORITES_IN_TIMESPAN_ID);
+          notes.add(new Object[] {thisNote, numFavoritesInTimespan});
+        }
+        rs.close();
+        return notes.toArray(new Object[0][0]);
+      }
+    }
+  }
+
+  /** Gets a sql where clause that filters by school or course if school or course is not null */
+  private String getStatementToFilterBySchoolOrCourse(String school, String course) {
+    String filterStmt = "WHERE 1=1 ";
+    if (school != null && !school.isEmpty()) {
+      filterStmt += "AND `school`= '" + school + "' ";
+    }
+    if (course != null && !course.isEmpty()) {
+      filterStmt += "AND `course`= '" + course + "' ";
+    }
+    return filterStmt;
+  }
+
+  /** Gets the sql query to return trending notes possibly filtered by school and/or course */
+  private String getTrendingNotesStatement(String schoolAndCourseFilter) {
+    String stmt = 
+        "SELECT * "
+      + "FROM " + NOTES + " AS a "
+      + "LEFT JOIN (SELECT note_id, COUNT(*) AS " + NUM_FAVORITES_IN_TIMESPAN_ID + " "
+                  + "FROM " + FAVORITE_NOTES + " "
+                  + "WHERE date >= ? " 
+                  + "GROUP BY note_id) AS b " 
+      + "ON a.id=b.note_id ";
+    if (schoolAndCourseFilter != null) stmt += schoolAndCourseFilter;
+    stmt += "ORDER BY count DESC;";
+    return stmt;
+  }
+
+  /** Gets the date based on recency */
+  private Date getDateBasedOnRecency(Recency recency) {
+    Date date;
+    Calendar calendar = Calendar.getInstance();
+    switch(recency) {
+      case TODAY:
```

Comment:
I think you don't need this because if it doesn't match a specific case, it just goes to `default`

Reply:
Oh you're right, I wasn't a 100% sure if I should still include it because it was technically still a "case", but then again I think I don't have the `ALL_TIME` case in other places so it makes sense not to include this here.